src/observer/index.js

```diff
@@ -47,7 +47,7 @@ Observer.create = function (value, vm) {
   }
   var ob
   if (
-    value.hasOwnProperty('__ob__') &&
+    '__ob__' in value &&
```

Comment:
Thanks, but using `in` operator this would return `true` if the `__ob__` is defined on the prototype as well. We can use `Object.prototype.hasOwnProperty.call(value, '__ob__')`.